### ISSUE TITLE
Ava+React+Storybook+Typescript+Snapshots+Wallaby gets "Test without test file id"

### ISSUE THREAD

**Issue by tugend** [wallabyjs/public#925]
### Issue description or question
Hi. I think Wallaby is a great product, thank you. =D

I'm struggling with an issue though, and I hope you can help. 
I think it's a matter of whether or not I'm able to encapsulate the test funtion
in another module. 

I'm running a setup with React, Typescript, Storybook and Ava, where I 
want to write stories for React Storybook like follows.

```
// __stories__/button.tsx
const stories = storiesOf('Min knap med tests', module, this);

stories.add('med custom konfiguration', t =>
    <button onClick={action('Hello World')}>
      Hello world
   </button>
);

```

Then running AVA via cli I'm using a wrapper that replace the storybook module 

``` 
// facade.cli.js
const snapshot = function(name, story, module, that) {
    let test = require('ava');
    // https://github.com/mthuret/storybook-addon-specifications/issues/4
    test.serial.call(that, name, function(t) {
      globals.options.file = module.filename;
      let renderer = require('react-test-renderer');
      const tree = renderer.create(story).toJSON();

      t.snapshot(tree);
    });
};

const storiesOf = function (description, module, that) {
  let api: any =  {};
  let story;
  api.add = api.addWithInfo = (name, func) => {
    story = func();
    snapshot(name, story, module, that);
    return api;
  };
  return api;
} as any;

```

The code works when run on the commandline, but when I run wallaby something goes wrong. it seems wallaby matches the throw assertion to the wrapper file instead of the test file. I have of course tried several variations of the code above. 

```javascript
// Wallay.js Tests
​​Some of the failing tests can not be reported, check the wallaby console for more details​​
```

```javascript
// Wallaby.js Console
​​[Error] Test without test file id, most likely the test did not run, OR the test/test body is in the file that is not in the `tests` list in the configuration file { testFileIdCanNotBeDetermined: 'med custom konfiguration',​​
​​[Error]   id: 100004,​​
​​[Error]   log: ​​
​​[Error]    [ { message: 'Please check your code or --update-snapshots\n\n  expect(value).toMatchSnapshot()\n\n  Received value does not match stored snapshot 2.\n\n  - Snapshot\n  + Received\n\n  <button\n      onClick={[Function]}>\n  -   Goodbye world fooxxxxx\n  +   Goodbye world fooxxxxxq\n    </button>',​​
​​[Error]        stack: [] } ],​​
​​[Error]   filesInvolved: [ 'libs/storybook.ts' ] }​​
​​[Error] Test without test file id, most likely the test did not run, OR the test/test body is in the file that is not in the `tests` list in the configuration file { testFileIdCanNotBeDetermined: 'med standard konfiguration',​​
​​[Error]   id: 100005,​​
​​[Error]   log: undefined,​​
​​[Error]   filesInvolved: [ 'libs/storybook.ts' ] }​​
​​[Error] Test without test file id, most likely the test did not run, OR the test/test body is in the file that is not in the `tests` list in the configuration file { testFileIdCanNotBeDetermined: 'app button',​​
​​[Error]   id: 100006,​​
​​[Error]   log: ​​
​​[Error]    [ { message: 'Please check your code or --update-snapshots\n\n  expect(value).toMatchSnapshot()\n\n  Received value does not match stored snapshot 2.\n\n  - Snapshot\n  + Received\n\n  <a\n      className="button"\n      download={undefined}\n      href="javascript: void(0)"\n      onClick={[Function]}\n      tabIndex={null}\n      title={undefined}>\n  -   Hello World\n  +   Hello Worldx\n    </a>',​​
​​[Error]        stack: [] } ],​​
​​[Error]   filesInvolved: ​​
​​[Error]    [ 'libs/storybook.ts',​​
​​[Error]      'app/controls/button.ts',​​
​​[Error]      'app/component.ts' ] }​​
```

### Wallaby.js configuration file

```javascript

module.exports = function(wallaby) {
    return {

        files: [
            "setup-test.ts",
            "libs/**/*",
            "build/**/*.snap",
            "app/**/*.ts",
            "app/**/*.tsx",
            "!**/__tests__/**",
            "!**/__stories__/**.tsx",
        ],

        tests: [
            "app/**/__tests__/**/*.ts",
            "app/**/__tests__/**/*.tsx",
            "app/**/__stories__/**/*.ts",
            "app/**/__stories__/**/*.tsx",
            "libs/**/__tests__/**/*.ts",
            "libs/**/__tests__/**/*.tsx"
        ],

        env: {
            type: "node"
        },
        debug: true,

        testFramework: "ava",

        bootstrap: function(wallaby) {
            require('./setup-test');
        }
    };
};
```

### Code editor or IDE name and version
Visual Studio Code latest

### OS name and version

Windows 10

### Related issues
This might be related to https://github.com/wallabyjs/public/issues/124.

**Comment by ArtemGovorov** [comment]
It is indeed directly related to the https://github.com/wallabyjs/public/issues/124. Dynamic tests support was added for `mocha` and `jasmine`, but `AVA` needed some more work. I have just published the support [in core v1.0.337](wallabyjs.com/docs/intro/troubleshooting.html#forcing-wallabyjs-core-update).

Please note that even though wallaby should be able to run the tests for you without the error message, I'd recommend against such organisation of your tests, as it obscures the source of errors (and not specifically for wallaby, but in general).

For example, when an assertion error is raised, it is obviously raised on this line:
```
 t.snapshot(tree);
```
of your `facade.cli.js`. And while you'll be able to see the test name in the error message, the stack of the error will obviously not contain this line:
```
stories.add('med custom konfiguration', t =>
    <button onClick={action('Hello World')}>
      Hello world
   </button>
);
```
meaning that neither wallaby.js, nor AVA via CLI, nor other test runners, are able to display the line as an error for you (in the stack trace or inline error message). And it makes sense, because this:
```
stories.add('med custom konfiguration', t => ...
```
is just a returning a test data for the assertion, and not really a test as such (the test itself is dynamically composed in the `facade.cli.js`).

Hope it makes sense.

**Comment by tugend** [comment]
Thank you, it does make sense. Much appriciated. 
I'll try to find a better approach.

Happy New Year!
